### PR TITLE
Feature/column adding

### PR DESCRIPTION
**Pull request by danielkulikov**
Added support for new column additions to ODS/HDS. No restrictions are placed on the function itself to create the column, so it's up to the user's discretion if they need to e.g. hardcode some internal column names into it.

### CODE REVIEW

File: gcp_airflow_foundations/operators/gcp/hds/hds_merge_table_operator.py [hunk 1]
```diff
@@ -44,6 +44,8 @@ class MergeBigQueryHDS(BigQueryOperator):
     :type gcp_conn_id: str
     :param column_mapping: Column mapping
     :type column_mapping: dict
+    :param column_adding: Columna dding
```
Comment (eugenemiretsky) [inline gcp_airflow_foundations/operators/gcp/hds/hds_merge_table_operator.py:47]: missing an a

**Review by eugenemiretsky — CHANGES_REQUESTED** [review]
what is the type pf column_adding? 
A dictionary of "col_name" -> sql_function? 

Can we make it more explicit?
1) Document it
2) Maybe give it a more concrete type so it can get validated during config validation?

**Comment by danielkulikov** [comment]
> what is the type pf column_adding? A dictionary of "col_name" -> sql_function?
> 
> Can we make it more explicit?
> 
> 1. Document it
> 2. Maybe give it a more concrete type so it can get validated during config validation?

I was going by how the column_casting was added - and it has some validation for it, but not for the actual types/dictionary keys expected. It should be:
      {"function": "SQL_FUNCTION", "output_type": BigQuery datatype}

I think we can just add explicit validation for both the casting and adding to make sure the dict is in the right form...what do you think? Creating configs for them will make configuration files extremely long and messy (esp w/ current use-cases)

**Comment by eugenemiretsky** [comment]
another random thought: 
Should we call it "new_colum_udfs"(user defined functions). Seems more appropriate

**Review by MichailParaskevopoulos — APPROVED** [review]

**Review by eugenemiretsky — APPROVED** [review]

**Pull request merged**